agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -241,23 +241,33 @@ private double getRequestDurationAvg() {
 
   /**
    * e.g. endpointUrl 'https://westus-0.in.applicationinsights.azure.com/v2.1/track' host will
-   * return 'westus-0.in.applicationinsights.azure.com'
+   * return 'westus-0'
    */
   static String getHost(String endpointUrl) {
```

Comment:
since this method is not called often, let's use the simpler (and less performant) `new URL(endpointUrl).getHost()` and then parse out the short name from there

Reply:
getHost is called every 15 mins for network statsbeat. 

public URL(URL context, String spec, URLStreamHandler handler) is not less performant if you checked their implementation though.

Reply:
constructing a new URL is expensive relative to not (which is why our `UrlParser` class exists), but every 15 minutes is no problem to use URL

Reply:
url.getHost doesn't do what we need. 

"https://fakehost-1.example.com" will return fakehost-1.example.com.

Reply:
is this method the same as `getGeoWithoutStampSpecific` from #2082?

Reply:
it's different.  this method return "westus-0", the other is returning region without the stamp specific "westus".

Reply:
got it, probably would be nice to leverage one of them to implement the other (strip off the trailing stamp #)